AutoRest/Modelers/Swagger/SchemaBuilder.cs

```diff
@@ -38,9 +38,12 @@ public override IType BuildServiceType(string serviceTypeName)
             _schema = Modeler.Resolver.Unwrap(_schema);
 
             // If primitive type
-            if ((_schema.Type != null && _schema.Type != DataType.Object) ||
-                _schema.AdditionalProperties != null)
+            if (_schema.Type != null && _schema.Type != DataType.Object || ( _schema.AdditionalProperties != null && _schema.Properties.IsNullOrEmpty() ))
             {
+                // Notes: 
```

Comment:
Could you update the [documentation here](https://github.com/Azure/autorest/blob/master/Documentation/defining-clients-swagger.md#dictionaries) too? Doesn't necessarily have to be with this PR if you just want to merge this as-is, but just mentioning it so that we don't forget to do it later